kotlin/src/com/google/idea/blaze/kotlin/sync/BlazeKotlinSyncPlugin.java

```diff
@@ -0,0 +1,252 @@
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.ModifiableRootModel;
+import com.intellij.openapi.roots.OrderRootType;
+import com.intellij.openapi.roots.impl.libraries.ProjectLibraryTable;
+import com.intellij.openapi.roots.libraries.Library;
+import com.intellij.openapi.roots.libraries.LibraryTable;
+import com.intellij.openapi.vfs.JarFileSystem;
+import com.intellij.openapi.vfs.VfsUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.kotlin.cli.common.arguments.CommonCompilerArguments;
+import org.jetbrains.kotlin.config.LanguageVersion;
+import org.jetbrains.kotlin.idea.compiler.configuration.KotlinCommonCompilerArgumentsHolder;
+import org.jetbrains.kotlin.idea.configuration.KotlinJavaModuleConfigurator;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import static com.google.idea.blaze.kotlin.BlazeKotlin.COMPILER_WORKSPACE_NAME;
+
+
+public class BlazeKotlinSyncPlugin extends BlazeKotlinBaseSyncPlugin {
+    @Override
+    public Collection<SectionParser> getSections() {
+        return BlazeKotlinSections.PARSERS;
+    }
+
+    /**
+     * Ensure the plugin is enabled and that the language version is as intended. The actual installation of the plugin should primarily be handled by
+     * {@link AlwaysPresentKotlinSyncPlugin}
+     */
+    @Override
+    public void updateProjectSdk(
+            Project project,
+            BlazeContext context,
+            ProjectViewSet projectViewSet,
+            BlazeVersionData blazeVersionData,
+            BlazeProjectData blazeProjectData
+    ) {
+        if (!blazeProjectData.workspaceLanguageSettings.isLanguageActive(LanguageClass.KOTLIN)) {
+            return;
+        }
+        if (PluginUtils.isPluginInstalled(BlazeKotlin.PLUGIN_ID)) {
+            PluginUtils.installOrEnablePlugin(BlazeKotlin.PLUGIN_ID);
+            maybeUpdateCompilerConfig(project, projectViewSet);
+        }
+    }
+
+    /**
+     * Update the compiler settings of the project if needed. The language setting applies to both the api version and the language version. Blanket setting
+     * this project wide is fine. The rules should catch incorrect usage.
+     */
+    private static void maybeUpdateCompilerConfig(Project project, ProjectViewSet projectViewSet) {
+        LanguageVersion languageLevel = BlazeKotlinSections.getLanguageLevel(projectViewSet);
+        String languageLevelVersionString = languageLevel.getVersionString();
+        CommonCompilerArguments settings = (CommonCompilerArguments) KotlinCommonCompilerArgumentsHolder.Companion.getInstance(project).getSettings().unfrozen();
+        boolean updated = false;
+        if (settings.getApiVersion() == null || !settings.getApiVersion().equals(languageLevelVersionString)) {
```

Comment:
This seems to be `settings.apiVersion` in 2017.2, but `settings.getApiVersion()` in 2017.3, we'll need an sdkcompat for this.